tasks/libs/junit_upload.py

```diff
@@ -150,14 +147,32 @@ def junit_upload_from_tgz(junit_tgz, codeowners_path=".github/CODEOWNERS"):
         # for each unpacked xml file, split it and submit all parts
         # NOTE: recursive=True is necessary for "**" to unpack into 0-n dirs, not just 1
         xmls = 0
+        processes = []
+        tempdirs = []
         for xmlfile in glob.glob(f"{unpack_dir}/**/*.xml", recursive=True):
             if not os.path.isfile(xmlfile):
                 print(f"[WARN] Matched folder named {xmlfile}")
                 continue
             xmls += 1
-            with tempfile.TemporaryDirectory() as output_dir:
-                written_owners, flavor = split_junitxml(xmlfile, codeowners, output_dir)
-                upload_junitxmls(output_dir, written_owners, flavor, xmlfile.split("/")[-1], process_env, tags)
+            output_dir = tempfile.TemporaryDirectory()
+            written_owners, flavor = split_junitxml(xmlfile, codeowners, output_dir.name)
+            processes.extend(
+                create_upload_junitxmls_processes(
+                    output_dir.name, written_owners, flavor, xmlfile.split("/")[-1], process_env, tags
+                )
+            )
+            tempdirs.append(output_dir)
+
+        # wait for the processes created to finish
+        for process in processes:
+            exit_code = process.wait()
+            if exit_code != 0:
+                raise subprocess.CalledProcessError(exit_code, DATADOG_CI_COMMAND)
+
+        # ensure the temporary directories created for each xml files are cleaned up
+        for dir in tempdirs:
```

Comment:
According to the documentation the cleanup will in all cases happen when the "temporarydirectory" object is released (worse case is process exit).
But I agree that this would be cleaner, let me update the code to reflect that

Reply:
@chouquette should be better with the latest commit